lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -95,8 +100,31 @@ class Sapui5MavenSnapshotResolver extends AbstractResolver {
 		}
 		const gav = metadata.gav.split(":");
 		let pkgName = metadata.npmPackageName;
-		if (!this._sources) {
+
+		// Use "npm-dist" artifact by default
+		let classifier;
+		let extension;
+		if (this._sources) {
+			// Use npm-sources artifact if sources are requested
+			classifier = "npm-sources";
+			extension = "zip";
+		} else if (this._isLegacySnapshotVersion) {
+			// For legacy versions < 1.116.0-SNAPSHOT where npm-dist artifact is not
+			// yet available, use "default" JAR
+			classifier = null;
+			extension = "jar";
+
+			// Add "prebuilt" suffix to package name
 			pkgName += "-prebuilt";
+		} else {
+			// Use "npm-dist" artifact by default
+			classifier = "npm-dist";
+			extension = "zip";
+
+			// Add "-prebuilt_dist" suffix to package name
+			// We can't use "-prebuilt" since old ui5-project versions might continue to download
+			// and store the default JAR under that name too
+			pkgName += "-prebuilt_dist";
```

Comment:
I wonder whether this is really the best solution. This is not the actual package name anymore.
Maybe the installer should rather have the knowledge on how to name the folders, because that's what is important here. The package name is actually not changing.

Reply:
Matthias and I found a potential solution to this. With the latest commit, we check whether the found "-prebuilt" package actually contains a package.json (which is only the case for the new npm-dist.zip artifact). In case no package.json is found, the installed package is overwritten with a newly installed npm-dist.zip package 👍

Reply:
Awesome!